### ISSUE TITLE
Refactor existing code

### ISSUE THREAD

**Issue by asvae** [asvae/vue-book#3]
Branch: https://github.com/asvae/vue-component-tree/tree/refactor

**Comment by davidmoshal** [comment]
@asvae I updated my Vue+Typescript+Poi starter:

https://github.com/davidmoshal/vue-typescript-poi-starter

it's really very easy to get started, the build boilerplate essentially disappears, and do dozens of explicity babel-related plugins - POI handles all of that for you, and manages Webpack 4.

**Comment by asvae** [comment]
I think it's good enough to try bundling.

I made dev server work on vue-cli-3. Also cut unneeded stuff. Bundling is dead though.

Next step I'll play with poi and ts-components. Any pointers are much appreciated, @davidmoshal.

![image](https://user-images.githubusercontent.com/5394573/40887846-5779d9de-6757-11e8-9b9a-4f8a55cf15e5.png)

**Comment by davidmoshal** [comment]
Ok, thanks, UI looks clean .

Searchable flat files much better than tree as discussed

one suggestion, allow easy navigation via up and down arrrows.

I updated my ts+poi starter repo to poi10 / webpack 4. Very easy to get started, minimal config, build directory conf files not needed, etc.

**Comment by asvae** [comment]
Checked poi-starter.

Figured it has exactly the same problem as vue-cli3.
Out of the box it gives 2 configs: `dev server` and `production build`. But apart from these 2 I also need `library build`. Library build is quite easy to implement on vue-cli2 or vanilla webpack. I've no idea how to implement it on vue-cli3 or poi. I tried to compile lib. Spent half a day. No result. You kinda have to start on top of existing config and chisel unneeded parts one by one. 

I think to rollback to vue-cli2. Not too fun nor fancy, but easiest solution to make things work.

BTW, If you have things to say and are OK with chat, here's my discord: asva#6179.
